python/cuml/fil/fil.pyx

```diff
@@ -325,17 +332,26 @@ cdef class ForestInference_impl():
                                       " using a Classification model, please "
                                       " set `output_class=True` while creating"
                                       " the FIL model.")
-        cdef uintptr_t X_ptr
-        X_m, n_rows, n_cols, dtype = \
-            input_to_cuml_array(X, order='C',
-                                convert_to_dtype=np.float32,
-                                safe_dtype_conversion=safe_dtype_conversion,
-                                check_dtype=np.float32)
-        X_ptr = X_m.ptr
-
         cdef handle_t* handle_ =\
             <handle_t*><size_t>self.handle.getHandle()
 
+        cdef uintptr_t X_ptr
+        cdef size_t n_rows, n_cols
+        cdef vector[column_view] cols
+        if type(X) == cudf.DataFrame:
```

Comment:
Is a single cuDF column also supported? This is likely not very interesting for FIL, but can be relevant for other use cases.